### PR TITLE
feat(eve): support dynamic remote subagent maps

### PR DESCRIPTION
**Pull request by OwenKephart**
### Summary

Dynamic subagent resolvers can now return a named map of `defineRemoteAgent()` definitions, exposing independently callable `<resolver>__<key>` tools. This provides a good foundation for the subsequent PR, but also would be conceptually useful for other sorts of remote agent registries.

### Validation

Focused dynamic-subagent, remote dispatch, cancellation, and public-definition coverage passed, including a regression for an entry named `kind`; `pnpm typecheck`, `pnpm docs:check`, and `pnpm guard:invariants` also passed.

### Checklist

- [x] This change was requested or approved by a maintainer
- [x] I ran the relevant checks from `CONTRIBUTING.md`
- [x] I added tests and documentation where relevant
- [x] I added a changeset if this touches the published `eve` package
- [x] DCO sign-off passes for every commit (`git commit --signoff`)

### Diff size

**Docs** — 2 files · `+34 / -0`

Documents the new map return form and records its patch release note.

**Implementation** — 8 files · `+186 / -73`

Extends the dynamic-subagent selection model and lifecycle lookup, plus the required retained public capability metadata.

**Tests** — 4 files · `+196 / -1`

Covers map naming and identity, including the `kind` key edge case, turn-level replacement, remote resolver lookup, public typing, and the retained extension API example.

### CODE REVIEW

File: packages/eve/src/context/dynamic-subagent-lifecycle.ts [hunk 1]
```diff
@@ -25,7 +26,66 @@ import { toErrorMessage } from "#shared/errors.js";
...
+function isDynamicSubagentMapSelection(
+  value: DurableDynamicSubagentResolverSelection,
+): value is Readonly<Record<string, DurableDynamicSubagentSelection>> {
+  return (
+    typeof value === "object" &&
+    value !== null &&
+    !Object.hasOwn(value, "kind") &&
+    Object.values(value).every(
+      (entry) => entry === null || (typeof entry === "object" && entry.kind === "remote"),
+    )
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply packages/eve/src/context/dynamic-subagent-lifecycle.ts]: Fixed in 0c4789803. Map detection now treats only actual `kind: "remote" | "subagent"` durable selections as single selections, so a map entry named `kind` remains a keyed map entry. Added regression coverage for dispatch and lookup of `researcher__kind`.
